keps/sig-apps/3715-elastic-indexed-job/README.md

```diff
@@ -0,0 +1,934 @@
+The controller already counts the number of completed indexes within the range defined by `spec.completions` and
+uses that to evaluate if the job completed successfully and to set the `status.Succeeded` count. This is not impacted
+if `spec.completions` changes over time. Morever, on scale down, the controller also automatically removes any indexes
+outside the range. 
+
+
+As mentioned above, pod failures may have happened to indexes that are outside the range after `spec.completions`
+is updated. Those failures will still count against the backoffLimit of the job and the field `status.Failed` will
+not be decremented.
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `k8s.io/kubernetes/pkg/controller/job`: `1/10/2023` - `90%`
+- `k8s.io/kubernetes/pkg/apis/batch/validation`: `1/10/2023` - `96%`
+
+##### Integration tests
+
+For an `Indexed` Job test that:
+- `spec.completions` is only mutable if equal to `spec.parallelism`
+- When scaled down, higher indexes are removed first.
+- Success, failure and completion semantics are as discussed above.
```

Comment:
Also test scale up.

Any other challenging scenarios for the controller you can think of?

Reply:
Scale down to zero.

Reply:
added.